pkg/plugins/policies/meshtrace/plugin/xds/configurer.go

```diff
@@ -0,0 +1,171 @@
+package xds
+
+import (
+	net_url "net/url"
+
+	envoy_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	envoy_trace "github.com/envoyproxy/go-control-plane/envoy/config/trace/v3"
+	envoy_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	tracingv3 "github.com/envoyproxy/go-control-plane/envoy/type/tracing/v3"
+	envoy_type "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/pkg/errors"
+
+	api "github.com/kumahq/kuma/pkg/plugins/policies/meshtrace/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/util/proto"
+	v3 "github.com/kumahq/kuma/pkg/xds/envoy/listeners/v3"
+)
+
+type Configurer struct {
+	Conf *api.MeshTrace_Conf
+
+	// Opaque string which envoy will assign to tracer collector cluster, on those
+	// which support association of named "service" tags on traces. Consumed by datadog.
+	Service     string
+	ClusterName string
+}
+
+var _ v3.FilterChainConfigurer = &Configurer{}
+
+func (c *Configurer) Configure(filterChain *envoy_listener.FilterChain) error {
+	if c.Conf.Backends[0] == nil {
+		return nil
+	}
+
+	backend := c.Conf.Backends[0]
+
+	return v3.UpdateHTTPConnectionManager(filterChain, func(hcm *envoy_hcm.HttpConnectionManager) error {
+		hcm.Tracing = &envoy_hcm.HttpConnectionManager_Tracing{}
+
+		if c.Conf.Sampling != nil {
+			hcm.Tracing.OverallSampling = &envoy_type.Percent{
+				Value: float64(c.Conf.Sampling.Overall), // how do we do defaults in this case (type default is 0 but we want Envoy default i.e. missing field)? a wrapper type and check null?
+			}
+			hcm.Tracing.ClientSampling = &envoy_type.Percent{
+				Value: float64(c.Conf.Sampling.Client),
+			}
+			hcm.Tracing.RandomSampling = &envoy_type.Percent{
+				Value: float64(c.Conf.Sampling.Random),
+			}
+		}
+
+		if c.Conf.Tags != nil {
+			hcm.Tracing.CustomTags = mapTags(c.Conf.Tags)
+		}
+
+		if backend.GetZipkin() != nil {
+			tracing, err := zipkinConfig(backend.Zipkin, c.ClusterName)
+			if err != nil {
+				return err
+			}
+			hcm.Tracing.Provider = tracing
+		}
+
+		if backend.GetDatadog() != nil {
+			tracing, err := datadogConfig(c.Service, c.ClusterName)
+			if err != nil {
+				return err
+			}
+			hcm.Tracing.Provider = tracing
+		}
+
+		return nil
+	})
+}
+
+func datadogConfig(serviceName, clusterName string) (*envoy_trace.Tracing_Http, error) {
+	datadogConfig := envoy_trace.DatadogConfig{
+		CollectorCluster: clusterName,
+		ServiceName:      serviceName,
```

Comment:
@Automaat did some stuff related to this recently. We should make sure the new policy doesn't introduce a regretion.

Reply:
I was about to ask if I should add the same logic for splitting. I'll add it here.